### PR TITLE
Core/ts version task 238 finalize event handler service

### PR DESCRIPTION
**Pull request by WilliamHawkingStephen**
## Check list  

- [x] All Merge Request pipelines should be passed.
- [x] Make sure all the changes are necessary and correct.
- [x] Make sure no needed change missing (all necessary changes are made).
- [x] Merge Request should contains only changes for one purpose (one feature or bug or refactoring).
- [ ] All necessary unit tests are written.
- [x] Critical changes should be tested locally (or on staging).

### CODE REVIEW

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 1]
```diff
@@ -25,21 +26,23 @@ export default class EventHandler
 
   public async process(): Promise<void> {
     // get latest read block
-    let latestReadBlock, events;
+    let latestReadBlock: Nullable<number>;
     const networks = await this.networkRepo.getNetworks();
     networks.forEach(
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:31]: we just need to read the latest read block of current network. loop is not needed.

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 2]
```diff
@@ -25,21 +26,23 @@ export default class EventHandler
 
   public async process(): Promise<void> {
     // get latest read block
-    let latestReadBlock, events;
+    let latestReadBlock: Nullable<number>;
     const networks = await this.networkRepo.getNetworks();
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:30]: we don't need network repo here. just set network entity. all related data is set in network entity. (provider, name, ...)

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 3]
```diff
@@ -25,21 +26,23 @@ export default class EventHandler
...
-        latestBlockNum,
-      );
+
+    const events = await this.blockchainNetwork.getEvents(
+      "*",
+      latestReadBlock ? latestReadBlock + 1 : latestBlockNum,
+      latestBlockNum,
+    );
     await this._handler(events as unknown as EVMEvents);
     await this.networkRepo.updateLatestReadBlock(
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:47]: bad input. ({name: string, latestReadBlock: number})

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 4]
```diff
@@ -2,48 +2,48 @@ import { IBlockchainEventHandlerService } from "@port/blockchain-event-handler-s
...
   }
 
   public async process(): Promise<void> {
     // get latest read block
-    let latestReadBlock, events;
-    const networks = await this.networkRepo.getNetworks();
-    networks.forEach(
-      provider =>
-        (latestReadBlock = this.networkRepo.getLatestReadBlock(provider.name)),
+    const networkName = Network.getNetwork(this.network)?.name;
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:29]: networkName is simply this.network

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 5]
```diff
@@ -2,48 +2,48 @@ import { IBlockchainEventHandlerService } from "@port/blockchain-event-handler-s
...
-      latestBlockNum as unknown as INetworkRepo.UpdateLatestReadBlock,
+
+    const events = await this.blockchainNetwork.getEvents(
+      "*",
+      latestReadBlock ? latestReadBlock + 1 : latestBlockNum,
+      latestBlockNum,
     );
+    await this._handler(events as unknown as EVMEvents);
+    await this.networkRepo.updateLatestReadBlock({
+      name: Network.name,
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:44]: name: this.network

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 6]
```diff
@@ -96,16 +97,17 @@ export default class EventHandler
...
         order: makerOffer,
-        status: OrderStatus.broadcasted,
+        status: OrderStatus.filled,
       });
       await this.FillAdapter.updateStatus({
         id: fillId,
-        status: FillStatus.broadcasted,
+        status: FillStatus.filled,
       });
+      fillDetails?.update({ txHash, status: FillStatus.filled });
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:110]: txHash is not saved in db.
pass it through updateStatus function

File: ts-version/src/domain/ports/repository/order.ts [hunk 7]
```diff
@@ -39,7 +44,7 @@ export namespace IOrderRepo {
     status: OrderStatus;
   };
 
-  export type ProcessClientOrderIdInput = {
+  export type InvalidatedOrderByClientOrderIdInput = {
```
Comment (hypersolo) [inline ts-version/src/domain/ports/repository/order.ts:47]: delete this dto

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 8]
```diff
@@ -123,23 +125,19 @@ export default class EventHandler
   }
 
   private async _processInvalidatedClientOrderId(
-    invalidateOrderInput: IOrderRepo.ProcessClientOrderIdInput,
+    invalidateOrderInput: IOrderRepo.InvalidatedOrderByClientOrderIdInput,
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:128]: IOrderRepo dtos are just for IOrderRepo.schema functions

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts (hunk 6 above)
Comment (WilliamHawkingStephen) [reply ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:110]: done

File: ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts [hunk 9]
```diff
@@ -88,58 +86,72 @@ export default class EventHandler
...
+        });
+        await orderRepository.updateStatus({
+          order: makerOffer,
+          status: OrderStatus.filled,
+        });
+        await fillRepository.updateStatus({
+          id: fillId,
+          status: FillStatus.filled,
+        });
+        await fillRepository.updateStatus({id: fillId, status: FillStatus.filled, txHash });
```
Comment (hypersolo) [inline ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:113]: fillRepositry.updateStatus function call is duplicated. delete first one.

Comment (WilliamHawkingStephen) [reply ts-version/src/domain/services/blockchain-event-handler/blockchain-event-handler.ts:109]: done

**Pull request merged**